client/http-client.js

```diff
@@ -1,14 +1,16 @@
 const qs = require('qs')
 const fetch = require('node-fetch')
+const HttpsProxyAgent = require('https-proxy-agent')
 
 module.exports = (clientConfig) => {
   // setup proxy agent in case a proxy is configured
   let agent = null
   if (clientConfig.agent) {
     agent = clientConfig.agent
   } else if (clientConfig.proxy) {
-    const HttpsProxyAgent = require('https-proxy-agent')
     agent = new HttpsProxyAgent(clientConfig.proxy)
+  } else if (process.env.HTTP_PROXY) {
+    agent = new HttpsProxyAgent(process.env.HTTP_PROXY)
```

Comment:
Maybe we should wrap that behavior in a custom proxy agent or do a PR against `https-proxy-agent` 😁
Currently we have that logic duplicated in a lot of services and there are definitely still issues in some of our setups.